evcouplings/complex/protocol.py

```diff
@@ -111,27 +315,373 @@ def _modify_segments(seg_list, seg_prefix):
+        ]
+    )
+
+    prefix = kwargs["prefix"]
+
+    # make sure input alignments
+    verify_resources(
+        "Input alignment does not exist",
+        kwargs["first_alignment_file"], kwargs["second_alignment_file"]
+    )
 
-    outcfg = {
-        "alignment_file": None,  # TODO: specify
-        "focus_mode": True,
-        "focus_sequence": None,  # TODO: specify
-        "segments": [s.to_list() for s in segments_complex],
-        # optional but good to have:
-        "num_sites": None,
-        "num_sequences": None,
-        # "effective_sequences": n_eff # TODO: could compute this like in align stage
-        # TODO: there are more outputs that we could add here (not mandatory),
-        # e.g. single column frequencies in concatenated alignment
-    }
+    # make sure output directory exists
+    create_prefix_folders(prefix)
+
+    def _load_monomer_info(annotations_file,
+                           identities_file,
+                           target_sequence,
+                           alignment_file):
+
+        # read in annotation to a file and rename the appropriate column
+        annotation_table = read_species_annotation_table(annotations_file)
+
+        # This dictionary maps identifiers used to calculate pairing
+        # to identifiers used to index into the sequence alignment
+        # Make sure the target sequence is included in this dictionary
+        # TODO: fix this so that we don"t assume target sequence is the first sequence
+        id_to_header = {x: [x] for x in annotation_table.id}
+        id_to_header[target_sequence] = [Alignment.from_file(open(alignment_file)).ids[0]]
+
+        # read identity file
+        similarities = pd.read_csv(identities_file)
+        # create a pd.DataFrame containing the best hit in each organism
+        most_similar_in_species = most_similar_by_organism(similarities, annotation_table)
+
+        return most_similar_in_species , id_to_header
+
+    # load the information about each monomer alignment
+    most_similar_in_species_1, id_to_header_1 = _load_monomer_info(
+        kwargs["first_annotation_file"],
+        kwargs["first_identities_file"],
+        kwargs["first_focus_sequence"],
+        kwargs["first_alignment_file"]
+    )
+
+    most_similar_in_species_2, id_to_header_2 = _load_monomer_info(
+        kwargs["second_annotation_file"],
+        kwargs["second_identities_file"],
+        kwargs["second_focus_sequence"],
+        kwargs["second_alignment_file"]
+    )
+
+    # merge the two dataframes to get all species found in 
+    # both alignments
+    species_intersection = most_similar_in_species_1.merge(
+        most_similar_in_species_2,
+        how="inner",  # takes the intersection
+        on="species",  # merges on species identifiers
+        suffixes=("_1", "_2")
+    )
+
+    raw_alignment_file = prefix + "_raw.fasta"
+
+    # write the raw concatenated alignment
+    target_seq_id, target_seq_index = write_concatenated_alignment(
```

Comment:
as I wrote inside write_concatenated_alignment, it would be nicer if the function gives back the Alignment object, then don't have to reload and protocol can write it to file